evm/vm/block_info.py

```diff
@@ -0,0 +1,62 @@
+class BlockInfo():
```

Comment:
Wondering if it would be better to not use the name *block* and instead to call this something like

* VMContext
* ExecutionContext

I'm partially thinking about https://github.com/ethereum/py-evm/issues/262 and whether the changes in this pull request can help us fix that issue as well (but not in this PR)